saleor/graphql/shop/resolvers.py

```diff
@@ -6,7 +6,8 @@
 
 def resolve_available_shipping_methods(info, channel_slug: str, address):
     available = ShippingMethod.objects.filter(
-        channel_listings__channel__slug=channel_slug
+        shipping_zone__channels__slug=channel_slug,
+        channel_listings__channel__slug=channel_slug,
```

Comment:
Maybe it would be useful to wrap this into a `for_channel` queryset? We have the same pattern for Stock model and it's easy to reuse and read.